src/backend/gl/src/tex.rs

```diff
@@ -663,8 +663,10 @@ fn bind_read_fbo(gl: &gl::Gl, texture: NewTexture, level: t::Level, fbo: FrameBu
     if texture == NewTexture::Surface(0) {
         //Warning: assuming the back buffer
         unsafe {
+            let mut buffer : gl::types::GLint = 0;
+            gl.GetIntegerv(gl::DRAW_BUFFER, &mut buffer);
```

Comment:
do you know what is the current draw buffer/FBO here? I don't think we can rely on that. E.g. what if we are currently copying data from the default FBO into a user FBO, the DRAW_BUFFER would then be `gl::COLOR_ATTACHMENT1`, which is not expected here.